# Data analysis

## 1. Hypotheses

**H1:** Trait mindfulness scores (MAAS) will predict phone checking frequency.

**H2:** Inhibitory control (Go/No-Go correct responses) will predict phone checking frequency.

**H3:** Trait mindfulness scores (MAAS) will predict inhibitory control (Go/No-Go performance).

In [1]:
import pandas as pd
import os
from scipy import stats
from scipy.stats import spearmanr
import plotly.express as px
from plotly.subplots import make_subplots
import plotly.graph_objects as go
import statsmodels.formula.api as smf


In [2]:
os.listdir("/Users/remyfernando/Desktop/Noguchi Active/BSc Psychology/Year 3/Research Project/Project data")

['.DS_Store',
 'Qualtrics_raw_original.csv',
 'inhibition_scores.csv',
 'Pavlovia completed :04:03',
 'Psychopy raw CSVs',
 'Old data analysis',
 'Qualtrics_ready_for_analysis.csv',
 'Qualtrics_cleaned.csv',
 'Qualtrics_raw.csv']

In [27]:
# file_path = "/Users/remyfernando/Desktop/Noguchi Active/BSc Psychology/Year 3/Research Project/Project data/Qualtrics_ready_for_analysis.csv"
# df = pd.read_csv(file_path)
# df.head()

with open("public_data_for_analysis.csv", "r") as f:
    df = pd.read_csv(f)
df.head()


,gender,age,device,maas_score,ios_pickups,non_ios_avg_pickups,inhibition_score
0,Female,24,iPhone,3.27,46.0,NaN,45
1,Female,22,iPhone,4.00,88.0,NaN,58
2,Female,37,iPhone,3.73,44.0,NaN,57
3,Female,33,Samsung,4.60,NaN,53.0,53
4,Female,33,Android,4.40,NaN,57.0,58


## 2. Descriptive statistics

In [28]:
df.head()

,gender,age,device,maas_score,ios_pickups,non_ios_avg_pickups,inhibition_score
0,Female,24,iPhone,3.27,46.0,NaN,45
1,Female,22,iPhone,4.00,88.0,NaN,58
2,Female,37,iPhone,3.73,44.0,NaN,57
3,Female,33,Samsung,4.60,NaN,53.0,53
4,Female,33,Android,4.40,NaN,57.0,58


In [29]:
df[['maas_score', 'ios_pickups', 'non_ios_avg_pickups', 'inhibition_score']].describe()

,maas_score,ios_pickups,non_ios_avg_pickups,inhibition_score
count,31.000000,14.000000,17.000000,31.000000
mean,3.913871,124.642857,59.647059,48.258065
std,0.691504,65.822146,34.965950,14.461369
min,2.670000,0.000000,29.000000,22.000000
25%,3.400000,90.250000,36.000000,37.500000
50%,4.000000,130.500000,52.000000,46.000000
75%,4.330000,170.500000,57.000000,58.000000
max,5.930000,215.000000,165.000000,75.000000


In [ ]:
columns_to_analyse = ['maas_score', 'ios_pickups', 'non_ios_avg_pickups', 'inhibition_score']
descriptive_stats = {}
for col in columns_to_analyse:
    descriptive_stats[col] = {
        "mean": round(df[col].mean(), 2),
        "standard_deviation": round(df[col].std(), 2)
    }

descriptive_df = pd.DataFrame(descriptive_stats).T # Transpose the DataFrame to have columns as rows and vice versa
print(descriptive_df)


                       mean  standard_deviation
maas_score             3.91                0.69
ios_pickups          124.64               65.82
non_ios_avg_pickups   59.65               34.97
inhibition_score      48.26               14.46


## 3. Normality & Outlier Assessment

### maas_score

In [31]:
# Histogram

maas_fig = make_subplots(rows=1, cols=2, subplot_titles=['Distribution', 'Box Plot'])
maas_fig.add_trace(go.Histogram(x=df['maas_score'], name='MAAS'), row=1, col=1)
maas_fig.add_trace(go.Box(y=df['maas_score'], name='MAAS'), row=1, col=2)
maas_fig.update_layout(title='MAAS Scores', showlegend=False)
maas_fig.show()


### Inhibition

In [32]:
inhibition_fig = make_subplots(rows=1, cols=2)
inhibition_fig.add_trace(go.Histogram(x=df['inhibition_score'], name='inhibition_score'), row=1, col=1)
inhibition_fig.add_trace(go.Box(y=df['inhibition_score'], name='inhibition_score'), row=1, col=2)
inhibition_fig.update_layout(title='Distribution and Box Plot of inhibition_score', showlegend=False)
inhibition_fig.show()


### IOS pickups

In [33]:
z_pickups_fig = make_subplots(rows=1, cols=2)
z_pickups_fig.add_trace(go.Histogram(x=df['ios_pickups'], name='Distribution'), row=1, col=1)
z_pickups_fig.add_trace(go.Box(y=df['ios_pickups'], name='Z-ios_pickups'), row=1, col=2)
z_pickups_fig.update_layout(title='Distribution and Box Plot of ios_pickups', showlegend=False)
z_pickups_fig.show()


### Non - IOS pickups

In [34]:
z_pickups_fig = make_subplots(rows=1, cols=2)
z_pickups_fig.add_trace(go.Histogram(x=df['non_ios_avg_pickups'], name='Distribution'), row=1, col=1)
z_pickups_fig.add_trace(go.Box(y=df['non_ios_avg_pickups'], name='Z-non_ios_pickups'), row=1, col=2)
z_pickups_fig.update_layout(title='Distribution and Box Plot of non_ios_avg_pickups', showlegend=False)
z_pickups_fig.show()


### Normality test

In [35]:
# Normality tests
columns_to_analyse = ['maas_score', 'ios_pickups', 'non_ios_avg_pickups', 'inhibition_score']

normality_results = {}
for col in columns_to_analyse:
    stat, p_value = stats.shapiro(df[col].dropna())
    normality_results[col] = {
        "skewness": round(df[col].skew(), 2),
        "kurtosis": round(df[col].kurtosis(), 2),
        "shapiro_statistic": round(stat, 4),
        "shapiro_p_value": round(p_value, 4),
        "normal": p_value > 0.05
    }
normality_df = pd.DataFrame(normality_results).T
print(normality_df)

                    skewness kurtosis shapiro_statistic shapiro_p_value normal
maas_score               0.4     1.09            0.9574          0.2487   True
ios_pickups            -0.36    -0.73            0.9498          0.5571   True
non_ios_avg_pickups     1.95     4.33            0.7743          0.0009  False
inhibition_score        0.06     -0.6            0.9746          0.6528   True


### Normality & Outlier Assessment Conclusion

Descriptive statistics and Shapiro-Wilk tests were conducted for each variable prior to analysis.

MAAS scores were approximately normally distributed (M = 3.91, SD = 0.69, skewness = 0.40, kurtosis = 1.09), confirmed by a non-significant Shapiro-Wilk test (W = 0.957, p = .249).

iOS pickup frequency was also normally distributed (M = 124.64, SD = 65.82, skewness = -0.36, kurtosis = -0.73), with a non-significant Shapiro-Wilk test (W = 0.950, p = .557).

Inhibition scores were normally distributed (skewness = 0.06, kurtosis = -0.60), confirmed by a non-significant Shapiro-Wilk test (W = 0.975, p = .653).

Non-iOS average pickups violated the assumption of normality (M = 59.65, SD = 34.97, skewness = 1.95, kurtosis = 4.33), with a significant Shapiro-Wilk test (W = 0.774, p = .001). Visual inspection of the distribution confirmed positive skew. One outlier was identified but retained as the Z-score fell within the acceptable ±3 range (Osborne & Overbay, 2004), and removal did not resolve the normality violation. Given the correlational design, slight deviations from normality were considered tolerable and analyses proceeded accordingly.

### Z-Score Standardisation

Phone pickup scores were standardised using Z-scores to account for systematic differences in pickup frequency between device types (iPhone vs Android/Samsung). iPhone users' pickup data comes from iOS screen time tracking, while Android and Samsung users self-reported daily pickups — these different measurement methods introduce scale differences. By calculating Z-scores separately within each device group, scores are placed on a consistent scale, allowing valid comparisons across all participants.

In [36]:
# Create a new column for z-scores and assign values based on device type
df['z_pickups'] = None
df.loc[df['ios_pickups'].notna(), 'z_pickups'] = stats.zscore(df['ios_pickups'].dropna())
df.loc[df['non_ios_avg_pickups'].notna(), 'z_pickups'] = stats.zscore(df['non_ios_avg_pickups'].dropna())

## 4. Inferential analysis

In [37]:
# Convert z_pickups to numeric
print(df[['z_pickups', 'maas_score', 'inhibition_score']].dtypes)

df["z_pickups"] = pd.to_numeric(df["z_pickups"], errors='coerce')

z_pickups            object
maas_score          float64
inhibition_score      int64
dtype: object


### Multile regression
Maas score and Inhibition as predictors with phone pickus as outcome

In [38]:
# Multiple regression, 
model1 = smf.ols('z_pickups ~ maas_score + inhibition_score', data=df).fit()
print(model1.summary())

                            OLS Regression Results                            
Dep. Variable:              z_pickups   R-squared:                       0.124
Model:                            OLS   Adj. R-squared:                  0.061
Method:                 Least Squares   F-statistic:                     1.980
Date:                Fri, 27 Feb 2026   Prob (F-statistic):              0.157
Time:                        16:39:43   Log-Likelihood:                -41.937
No. Observations:                  31   AIC:                             89.87
Df Residuals:                      28   BIC:                             94.18
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                       coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------------
Intercept            1.0669      1.050  

### Multiple Regression Results

A multiple regression was conducted with MAAS score and inhibition score as predictors of standardised phone pickup frequency (z_pickups).

The overall model was not statistically significant (R² = 0.124, Adj. R² = 0.061, F(2, 28) = 1.98, p = .157), suggesting that MAAS and inhibition scores together explained only 12.4% of the variance in phone checking behaviour.

MAAS score was not a significant predictor (β = 0.04, t = 0.15, p = .884). Inhibition score approached but did not reach significance (β = -0.03, t = -1.86, p = .073), with the negative coefficient suggesting that higher inhibitory control was associated with fewer phone pickups — a direction consistent with theoretical expectations.

The Durbin-Watson statistic (1.892) indicated no substantial autocorrelation in the residuals. The Omnibus test was significant (p = .043), indicating mild non-normality in the residuals, consistent with the normality violations identified during descriptive analysis.

In [39]:
fig1 = px.scatter(df, x='maas_score', y='z_pickups', trendline='ols', title='MAAS vs Z-Pickups')
fig1.show()

In [40]:
fig2 = px.scatter(df, x='inhibition_score', y='z_pickups', trendline='ols', title='Inhibition vs Z-Pickups')
fig2.show()

### Simple regression
With MAAS score as predictor and inhibition as outcome


In [41]:
# Simple regression
model2 = smf.ols('inhibition_score ~ maas_score', data=df).fit()
print(model2.summary())

                            OLS Regression Results                            
Dep. Variable:       inhibition_score   R-squared:                       0.178
Model:                            OLS   Adj. R-squared:                  0.150
Method:                 Least Squares   F-statistic:                     6.273
Date:                Fri, 27 Feb 2026   Prob (F-statistic):             0.0181
Time:                        16:39:45   Log-Likelihood:                -123.26
No. Observations:                  31   AIC:                             250.5
Df Residuals:                      29   BIC:                             253.4
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     13.7396     13.988      0.982      0.3

In [42]:
fig3 = px.scatter(df, x='maas_score', y='inhibition_score', trendline='ols', title='MAAS vs Inhibition')
fig3.show()

### Simple Regression Results

A simple linear regression was conducted with MAAS score as a predictor of inhibition score.

The model was statistically significant (R² = 0.178, Adj. R² = 0.150, F(1, 29) = 6.27, p = .018), with MAAS score explaining 17.8% of the variance in inhibition score.

MAAS score was a significant positive predictor of inhibition (β = 8.82, t = 2.51, p = .018), suggesting that higher trait mindfulness was associated with better inhibitory control.

The Durbin-Watson statistic (1.182) indicates some positive autocorrelation in the residuals, which should be interpreted with caution. However, the Omnibus test was non-significant (p = .976), suggesting residuals were approximately normally distributed.

### Correlation Table

In [43]:
relevant_columns = ['maas_score', 'inhibition_score', 'z_pickups']
correlation_matrix = df[relevant_columns].corr(method='spearman')
print(correlation_matrix)

                  maas_score  inhibition_score  z_pickups
maas_score          1.000000          0.401274  -0.006160
inhibition_score    0.401274          1.000000  -0.326669
z_pickups          -0.006160         -0.326669   1.000000


In [44]:

correlations = {}
for col in relevant_columns:
    for col2 in relevant_columns:
        if col != col2:
            corr, p = spearmanr(df[col], df[col2])
            correlations[(col, col2)] = (corr, p)
correlation_matrix = pd.DataFrame(correlations, index=['spearman_correlation', 'p_value']).T
print(correlation_matrix)

                                   spearman_correlation   p_value
maas_score       inhibition_score              0.401274  0.025263
                 z_pickups                    -0.006160  0.973762
inhibition_score maas_score                    0.401274  0.025263
                 z_pickups                    -0.326669  0.072870
z_pickups        maas_score                   -0.006160  0.973762
                 inhibition_score             -0.326669  0.072870


## 5. Conclusion

**H1 — Not supported.** MAAS score was not a significant predictor of phone checking frequency in the multiple regression (β = 0.04, p = .884).

**H2 — Not supported.** Inhibition score did not significantly predict phone checking frequency, though it approached significance (β = -0.03, p = .073), with the direction of effect consistent with the hypothesis.

**H3 — Supported.** MAAS score significantly predicted inhibitory control (β = 8.82, p = .018), suggesting that higher trait mindfulness is associated with better inhibitory control.